kite-data/kite-data-core/src/main/java/org/kitesdk/data/Datasets.java

```diff
@@ -31,25 +31,52 @@
 import org.kitesdk.data.spi.Registration;
 
 /**
- * <p>Convenience methods for working with {@link Dataset} instances.</p>
- *
+ * Methods for working with {@link Dataset} instances.
+ * <p>
+ * <b>URIs</b><br />
+ * All methods require a URI that identifies a dataset, view, or
+ * repository. The URI must begin with the scheme {@code dataset:}, 
+ * {@code view:}, or {@code repo:}. The remainder of the URI is 
+ * implementation specific, depending on the dataset scheme.
+ * <p>
+ * For example, the URI {@code dataset:hive:/movies/ratings} 
+ * references a Hive dataset named <i>ratings</i> in the
+ * <i>movies</i> namespace.
+ * <p>
+ * The URI {@code view:hdfs:/user/myfiles/movies/ratings?year=2015&month=3}
+ * references a view of an HDFS dataset named <i>ratings</i> in the
+ * <i>movies</i> namespace in the {@code /user/myfiles} path. The view 
+ * is filtered to include records from only March, 2015.
+ * <p>
+ * See <a href="http://kitesdk.org/docs/current/URIs.html">Dataset and View
+ * URIs</a>.
+ * <p>
+ * <b>Entities</b><br />
+ * <i>Entities</i> are analagous to <i>records</i> in database terminology.
+ * The term is used in the API to emphasize that an entity can include not
+ * only primitive objects, but also complex objects such as hash maps. 
+ *  
  * @since 0.8.0
  */
 public class Datasets {
 
   /**
    * Load a {@link Dataset} or {@link View} for the given {@link URI}.
+   * {@code load} verifies that the dataset exists, retrieves the dataset's
```

Comment:
It is sufficient to say that it loads the dataset and retrieves its metadata. I wouldn't say that this verifies that a dataset exists or that it verifies that you can communicate with its services. In the first case, it doesn't specifically check. It just throws an exception if there is no dataset and it can't proceed.

For the part about communicating with its services, we don't really think of a dataset as something that has services. I think you mean that it verifies the communication with any services needed to access the data, but that isn't strictly true. We could load information from the Hive MetaStore without necessarily verifying that we can communicate with HDFS.